bat-utils/lib/hapi-auth-whitelist.js

```diff
@@ -35,11 +37,17 @@ exports.ipaddr = (request) => {
   // Since it is easy to forge an X-Forwarded-For field the given information should be used with care.
   // The last IP address is always the IP address that connects to the last proxy, which means it is the most reliable source of information.
 
-  const forwardedFor = request.headers['x-forwarded-for']
+  const { headers } = request
+  const forwardedFor = headers['x-forwarded-for']
+  const token = headers['fastly-token']
   if (forwardedFor) {
-    const shift = forwardedIPShift()
+    if (fastlyTokens.length && !braveHapi.isSimpleTokenValid(fastlyTokens, token)) {
```

Comment:
can we fail if we're in production and the env var is not passed?

Reply:
no, defaults to an empty array above: https://github.com/brave-intl/bat-ledger/pull/631/files#diff-d3f10f89496dc0476f73b80d818352e5R9

Reply:
sorry I wasn't clear. can you please make it fail in that case

Reply:
yeah, sure